sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/ConstantFoldingSuite.scala

```diff
@@ -437,6 +437,21 @@ class ConstantFoldingSuite extends PlanTest {
       Optimize.execute(oneRowScalarSubquery),
       oneRowScalarSubquery)
   }
+
+  test("Current time functions are constant folded") {
```

Comment:
I think what we need is an analyzer test to make sure we can use current_time in functions that require folding input, like `rand`. How about we move the test to `ExpressionTypeCheckingSuite`?